File: utils/sanitize.js
export const sanitize = (str) => {
  // Remove script tags
  let sanitized = str.replace(/<script.*?>.*?<\/script>/gi, "");
  // Escape special HTML characters
  sanitized = sanitized
    .replace(/&/g, "&amp;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;")
    .replace(/"/g, "&quot;")
    .replace(/'/g, "&#39;");
  return sanitized;
};

File: utils/backend/utils/sanitize.go
package utils

import (
	"regexp"
	"strings"
)

// Sanitize removes all HTML tags and escapes angle brackets.
func Sanitize(input string) string {
	// Remove all HTML tags
	re := regexp.MustCompile(`(?i)<.*?>`)
	sanitized := re.ReplaceAllString(input, "")
	// Escape any remaining angle brackets
	sanitized = strings.ReplaceAll(sanitized, "<", "&lt;")
	sanitized = strings.ReplaceAll(sanitized, ">", "&gt;")
	return sanitized
}
